### ISSUE TITLE
Can't use option with type array

### ISSUE THREAD

**Issue by artofhuman** [hanami/cli#12]
``` ruby
#!/usr/bin/env ruby

require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'hanami-cli', github: 'hanami/cli'
end

require 'hanami/cli'

module HanamiTestCLI
  extend Hanami::CLI::Registry

  class TestCommand < Hanami::CLI::Command
    option :only, type: :array

    def call(**options)
      puts options
    end
  end

  register 'test', TestCommand
end

Hanami::CLI.new(HanamiTestCLI).call
```
```
› ruby hanami-cli-bug.rb test --only foo bar
{:only=>"foo"}
```

I think `thor` allow to pass options separated by space as an array, but `hanami-cli` no, @jodosha WDYT?

**Comment by davydovanton** [comment]
hey, thanks for report. I think it helpful feature. Could you create PR for this?

**Comment by AlfonsoUceda** [comment]
@artofhuman thanks for this, just one question, what are you trying to achieve? I'd like to see a real example.

**Comment by artofhuman** [comment]
@AlfonsoUceda 
I tried to integrate `hanami-cli` to `hanami-scaffold` and found that new `hanami-cli` can`t work with commands like this https://github.com/davydovanton/hanami-scaffold/blob/master/lib/hanami/scaffold.rb#L35

For example, old commands have `array` option
https://github.com/davydovanton/hanami-scaffold/blob/master/lib/hanami/scaffold.rb#L24-L25

I think we can use this code https://github.com/artofhuman/hanami-scaffold/commit/92bded8bf201f394b49d67f7bb48d875107aeeb1 to reproduce this case

**Comment by AlfonsoUceda** [comment]
@artofhuman it makes sense then, OptionParser support array options and I think we should implement it. I'll take care of this feature.

**Comment by AlfonsoUceda** [comment]
@artofhuman I forgot about values option, please can you try it?

**Comment by AlfonsoUceda** [comment]
https://github.com/hanami/cli/commit/aa2b0842e9546b88250dc1943df3c72f6d907e33

**Comment by AlfonsoUceda** [comment]
closed by #14

Hi @artofhuman after almost one year, this is done, at the end we did it ;)

**Comment by artofhuman** [comment]
Cool 👍
